src/node.cpp

```diff
@@ -0,0 +1,45 @@
+/**
+ * @file      node.cpp
+ * @brief     Node class definitions
+ * @author    Julian van Doorn
+ * @license   See LICENSE
+ */
+
+#include "node.hpp"
+
+namespace Pathfinding {
+Node::Node() {
+}
+
+Node::Node(uint32_t id) : id(id) {
+}
+
+Node::Node(uint32_t id, Node **edges, uint32_t edgesCount) : id(id), edges(edges), edgesCount(edgesCount) {
+}
+
+void Node::getEdges(Node ***edges, uint32_t *edgesCount) const {

```

Comment:
edgesCount should be a reference, not a pointer

Perhaps create a type def for edges and give it as a reference